### PR TITLE
4 convert the datum postprocessing class

### PR DESCRIPTION
**Pull request by matdenkas**
Just build and run:
`docker exec flare-backend python3 -m pytest -vv -s`

### CODE REVIEW

File: backend/PostProcessing/PostProcessingClasses/LinearInterpolation.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#-------------------------------
+# 
+#
+#Imports
+from PostProcessing.IPostProcessing import IPostProcessing
+from pandas import DataFrame, date_range
+import numpy as np
+from datetime import timedelta
+
+class LinearInterpolation(IPostProcessing):
```
Comment (abeasley1722) [inline backend/PostProcessing/PostProcessingClasses/LinearInterpolation.py:17]: My main question is whether this is meant to replace the post-processing class used for interpolating between measurements and predictions. If so, how will this be implemented ? Say we have three models that predict at 12-hour, 24-hour, and 48-hour intervals. You would need to take one prediction from each model, along with the last available measurement from the measured data, and create a data frame containing these data points. Then, you would generate null data points between the values and pass this data frame into this post-processing class. When exactly would this process occur?


A visual eaxample of what I mean:

| Date               | Value            |
|--------------------|------------------|
| 2024-11-20 00:00   | measurement      |
| 2024-11-20 01:00   | null             |
| 2024-11-20 02:00   | null             |
| 2024-11-20 03:00   | null             |
| 2024-11-20 04:00   | null             |
| 2024-11-20 05:00   | null             |
| 2024-11-20 06:00   | null             |
| 2024-11-20 07:00   | null             |
| 2024-11-20 08:00   | null             |
| 2024-11-20 09:00   | null             |
| 2024-11-20 10:00   | null             |
| 2024-11-20 11:00   | null             |
| 2024-11-20 12:00   | 12 hr prediction |
| 2024-11-20 13:00   | null             |
| 2024-11-20 14:00   | null             |
| 2024-11-20 15:00   | null             |
| 2024-11-20 16:00   | null             |
| 2024-11-20 17:00   | null             |
| 2024-11-20 18:00   | null             |
| 2024-11-20 19:00   | null             |
| 2024-11-20 20:00   | null             |
| 2024-11-20 21:00   | null             |
| 2024-11-20 22:00   | null             |
| 2024-11-20 23:00   | null             |
| 2024-11-21 00:00   | 24 hr prediction |
| 2024-11-21 01:00   | null             |
| 2024-11-21 02:00   | null             |
| 2024-11-21 03:00   | null             |
| 2024-11-21 04:00   | null             |
| 2024-11-21 05:00   | null             |
| 2024-11-21 06:00   | null             |
| 2024-11-21 07:00   | null             |
| 2024-11-21 08:00   | null             |
| 2024-11-21 09:00   | null             |
| 2024-11-21 10:00   | null             |
| 2024-11-21 11:00   | null             |
| 2024-11-21 12:00   | null             |
| 2024-11-21 13:00   | null             |
| 2024-11-21 14:00   | null             |
| 2024-11-21 15:00   | null             |
| 2024-11-21 16:00   | null             |
| 2024-11-21 17:00   | null             |
| 2024-11-21 18:00   | null             |
| 2024-11-21 19:00   | null             |
| 2024-11-21 20:00   | null             |
| 2024-11-21 21:00   | null             |
| 2024-11-21 22:00   | null             |
| 2024-11-21 23:00   | null             |
| 2024-11-22 00:00   | 48 hr prediction |

File: backend/demo.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     backwards_interpolation[forward_cumulative_nan_mask] = np.nan
+
+     # We have to reindex the data back in, hopefully this reindex does nothing and Nans were in-placed correctly before this call.
+     #df = df.join(index=data.index, copy=False, fill_value=np.nan)
+     #df.loc[:, col_name] = backwards_interpolation
+     df.drop(columns=[col_name], inplace=True)
+     df = df.join(backwards_interpolation, how='outer')
+     
+     return df
+
```
Comment (abeasley1722) [inline backend/demo.py:63]: Other than what I mentioned everything looks well documented and thought out! I know you already have a ton on your plate so thank you for taking this over for me.

File: backend/PostProcessing/PostProcessingClasses/ImmediateArithmeticOperation.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    def post_process(self, data: DataFrame, op: str, left_col_key: str, value: float, out_col_key: str) -> DataFrame:
+        """The post processing in this file preforms an arithmetic operation on columns in a data frame by a constant.
+        By the index of the data frame!
+
+        Args:
+            data DataFrame: The DataFrame containing this collation of the data that has been collected.
+            op - the operation to preform (ex. addition)
+            left_col_key - The key the column that will be on the left side of the operation.
+            right_col_key - The key for the column on the right side of the operation.
```
Comment (lovelysandlonelys) [inline backend/PostProcessing/PostProcessingClasses/ImmediateArithmeticOperation.py:28]: the function defintion doesn't have a right_col_key parameter, it has a value: float parameter

File: backend/demo.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import pandas as pd
```
Comment (lovelysandlonelys) [inline backend/demo.py:1]: the demo.py file shouldn't be pushed to the repository?

**Review by lovelysandlonelys — CHANGES_REQUESTED** [review]
two super duper small things! tests all passed: 
image

**Pull request merged**